### PR TITLE
[refactor] use utility method of Objects to simplify code

### PR DESCRIPTION
**Pull request by caidezhi**
use utility method ( Objects.equals ) to rewrite some equals logic in code base.

### CODE REVIEW

File: maven-artifact/src/main/java/org/apache/maven/artifact/versioning/VersionRange.java [hunk 1]
```diff
@@ -19,12 +19,7 @@
  * under the License.
  */
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.WeakHashMap;
+import java.util.*;
```
Comment (olamy) [inline maven-artifact/src/main/java/org/apache/maven/artifact/versioning/VersionRange.java]: we do not use star import

File: maven-core/src/main/java/org/apache/maven/plugin/DefaultPluginDescriptorCache.java [hunk 2]
```diff
@@ -19,10 +19,7 @@
  * under the License.
  */
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (olamy) [inline maven-core/src/main/java/org/apache/maven/plugin/DefaultPluginDescriptorCache.java]: we do not use star import

File: maven-core/src/main/java/org/apache/maven/project/MavenProject.java [hunk 3]
```diff
@@ -22,15 +22,7 @@
...
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
-import java.util.Set;
+import java.util.*;
```
Comment (olamy) [inline maven-core/src/main/java/org/apache/maven/project/MavenProject.java]: we do not use star import

File: maven-core/src/main/java/org/apache/maven/project/artifact/DefaultMavenMetadataCache.java [hunk 4]
```diff
@@ -20,13 +20,7 @@
...
 
 import java.io.File;
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.LinkedHashMap;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (olamy) [inline maven-core/src/main/java/org/apache/maven/project/artifact/DefaultMavenMetadataCache.java]: we do not use star import

File: maven-core/src/main/java/org/apache/maven/toolchain/DefaultToolchain.java [hunk 5]
```diff
@@ -19,10 +19,7 @@
  * under the License.
  */
 
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Properties;
+import java.util.*;
```
Comment (olamy) [inline maven-core/src/main/java/org/apache/maven/toolchain/DefaultToolchain.java]: we do not use star import

File: maven-core/src/main/java/org/apache/maven/plugin/DefaultPluginDescriptorCache.java (hunk 2 above)
Comment (caidezhi) [reply maven-core/src/main/java/org/apache/maven/plugin/DefaultPluginDescriptorCache.java]: thanks for review.I have modified the code as your suggestion.

**Comment by rfscholte** [comment]
@caidezhi next time please run `mvn verify` on the project before creating a PR. You would have noticed that checkstyle would have complaint. I'll fix it on our master.

**Comment by caidezhi** [comment]
@rfscholte I am sorry that the pull request cause some inconveniences (forget that my IDEA setting is "-Dcheckstyle.skip=true"). I will follow the process in future. Many thanks for your help.

**Pull request merged**